engine/execution/state/bootstrap/bootstrap.go

```diff
@@ -94,27 +94,36 @@ func (b *Bootstrapper) IsBootstrapped(db *badger.DB) (flow.StateCommitment, bool
 	return commit, true, nil
 }
 
-func (b *Bootstrapper) BootstrapExecutionDatabase(db *badger.DB, commit flow.StateCommitment, genesis *flow.Header) error {
+func (b *Bootstrapper) BootstrapExecutionDatabase(
+	db *badger.DB,
+	rootSeal *flow.Seal,
+) error {
 
+	commit := rootSeal.FinalState
 	err := operation.RetryOnConflict(db.Update, func(txn *badger.Txn) error {
 
-		err := operation.InsertExecutedBlock(genesis.ID())(txn)
+		err := operation.InsertExecutedBlock(rootSeal.BlockID)(txn)
 		if err != nil {
 			return fmt.Errorf("could not index initial genesis execution block: %w", err)
 		}
 
+		err = operation.SkipDuplicates(operation.IndexExecutionResult(rootSeal.BlockID, rootSeal.ResultID))(txn)
```

Comment:
I wonder why this needs `SkipDuplicates`... Probably this is the only place we should be indexing the root result. If `SkipDuplicates` is really needed, it's likely a bug or duplication of logic.

Reply:
for dynamic bootstrapped EN, then rootSeal.BlockID != rootBlock.BlockID, so we need to index here. 
But for spork bootstrapped EN, then rootSeal.BlockID == rootBlock.BlockID, we would hit key already exist error when storing root result.